cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/batch/sources/DBSource.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.templates.etl.batch.sources;
+
+import co.cask.cdap.templates.etl.api.Property;
+import co.cask.cdap.templates.etl.api.StageConfigurer;
+import co.cask.cdap.templates.etl.api.batch.BatchSource;
+import co.cask.cdap.templates.etl.api.batch.BatchSourceContext;
+import co.cask.cdap.templates.etl.common.DBRecord;
+import co.cask.cdap.templates.etl.common.Properties;
+import com.google.common.base.Preconditions;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.lib.db.DBConfiguration;
+import org.apache.hadoop.mapreduce.lib.db.DBInputFormat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+/**
+ * Batch source to read from a Database table
+ */
+public class DBSource extends BatchSource<LongWritable, DBRecord> {
+  private static final Logger LOG = LoggerFactory.getLogger(DBSource.class);
+
+  @Override
+  public void configure(StageConfigurer configurer) {
+    configurer.setName("DBBatchSource");
+    configurer.setDescription("Batch source for database");
+    configurer.addProperty(new Property(Properties.DB.DB_DRIVER_CLASS, "Driver class to connect to the database", true));
+    configurer.addProperty(new Property(Properties.DB.DB_CONNECTION_STRING, "JDBC connection string including database name", true));
+    configurer.addProperty(new Property(Properties.DB.DB_TABLE_NAME, "Table name to import", true));
+    configurer.addProperty(new Property(Properties.DB.DB_COLUMNS, "Comma-separated list of columns to import", false));
+  }
+
```

Comment:
What do you mean by don't want to create it? We should never create a table in the source?